bundles/org.openhab.core.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttBrokerConnection.java

```diff
@@ -513,17 +582,19 @@ public boolean hasSubscribers() {
      */
     protected CompletableFuture<Boolean> subscribeRaw(String topic) {
         logger.trace("subscribeRaw message consumer for topic '{}' from broker '{}'", topic, host);
-        CompletableFuture<Boolean> future = new CompletableFuture<>();
-        try {
-            MqttAsyncClient client = this.client;
-            if (client != null && client.isConnected()) {
-                client.subscribe(topic, qos, future, actionCallback);
-            } else {
-                future.complete(false);
-            }
-        } catch (org.eclipse.paho.client.mqttv3.MqttException e) {
-            logger.info("Error subscribing to topic {}", topic, e);
-            future.completeExceptionally(e);
+        CompletableFuture<Boolean> future = new CompletableFuture<Boolean>();
+        final MqttAsyncClientWrapper client = this.client;
+        if (client != null && client.getState().isConnected()) {
+            client.subscribe(topic, qos, clientCallback).whenComplete((s, t) -> {
+                if (t == null) {
+                    future.complete(true);
+                } else {
+                    logger.info("Error subscribing to topic {}", topic, t);
```

Comment:
If this is an error, why not log an error?

Reply:
I changed the wording and upgraded to warn. error is too much IMO, this does not affect the funtion of the other topics or the connection.